src/react.js

```diff
@@ -151,18 +151,27 @@ function createContext () {
     }
   }
 
-  function differ (mappedProps, nextMappedProps) {
+  function differ (mappedProps, nextMappedProps, { objects = false } = {}) {
+    if (mappedProps === nextMappedProps) {
+      return false
+    }
+    if (!mappedProps || !nextMappedProps) {
+      return true
+    }
     for (let i in mappedProps) {
       if (mappedProps[i] !== nextMappedProps[i]) return true
     }
     for (let i in nextMappedProps) {
       if (!(i in mappedProps)) return true
     }
+    if (!objects && mappedProps !== nextMappedProps) {
+      return true
+    }
     return false
   }
 
-  return { Provider, Consumer, connect }
+  return { AtomContext, Provider, Consumer, connect, differ }
 }
 
-const { Provider, Consumer, connect } = createContext()
-module.exports = { Provider, Consumer, connect, createContext }
+const { AtomContext, Provider, Consumer, connect, differ } = createContext()
+module.exports = { AtomContext, Provider, Consumer, connect, differ, createContext }
```

Comment:
Do you think there's a way to not expose `AtomContext` directly to the user? Not a big fan of that

Reply:
Also why do we expose `createContext`?

Reply:
I don't see how not to expose it easily now that I've split the logic across 2 modules.

Perhaps you're right in that most likely nobody would use those exports. But you could use AtomContext for building your own hook and reaching it with useContext(AtomContext). And createContext, yeah, I guess there's a lot of yagni here, can't think of a reason why you'd need it. I added it when I was a bit confused about the new context api and thought that it's like a global singleton kind of thing, but turns out Consumer sees what Provider provides through the tree, so indeed exporting this low level thing isn't needed.

Reply:
One use case / edge case for createContext is if you want to use atom in some component, say a calendar widget, but it also renders nested children. This way, calendar widget could createContext() and use <Provider /> <Consumer /> such that it doesn't interfere with the main app's <Provider /> and <Consumers />.